lib/oxidized/model/asa.rb

```diff
@@ -27,27 +34,53 @@ class ASA < Oxidized::Model
     comment cfg
   end
 
-  cmd 'more system:running-config' do |cfg|
-    cfg = cfg.each_line.to_a[3..-1].join
-    cfg.gsub! /^: [^\n]*\n/, ''
-    # backup any xml referenced in the configuration.
-    anyconnect_profiles = cfg.scan(Regexp.new('(\sdisk0:/.+\.xml)')).flatten
-    anyconnect_profiles.each do |profile|
-  	  cfg << (comment profile + "\n" )
-   	  cmd ("more" + profile) do |xml|
-	      cfg << (comment xml)
-	    end
-    end
-    # if DAP is enabled, also backup dap.xml
-    if cfg.rindex(/dynamic-access-policy-record\s(?!DfltAccessPolicy)/)
-   	  cfg << (comment "disk0:/dap.xml\n")
-      cmd "more disk0:/dap.xml" do |xml|
-        cfg << (comment xml)
+  post do
+    if (@context == true)
+      # Multiple context mode
+      cmd 'changeto system' do |cfg|
+        cmd 'show running-config' do |systemcfg|
+          allcfg = "\n\n" + systemcfg + "\n\n"
+          contexts = systemcfg.scan(/^context (\S+)$/)
+          files = systemcfg.scan(/config-url (\S+)$/)
+          i = 0
+          contexts.each do |cont|
+            allcfg = allcfg + "\n\n---=== [ CONTEXT " + cont.join(" ") + " FILE " + files[i].join(" ") + " ] ===---\n\n"
+            cmd "more " + files[i].join(" ") do |cfgcontext|
+              allcfg = allcfg + "\n\n" + cfgcontext
+            end
+            i += 1
+          end
+          cfg = allcfg
+        end
+        cfg
+      end
+    else
+      # Single context mode
+      cmd 'more system:running-config' do |cfg|
+        cfg = cfg.each_line.to_a[3..-1].join
+        cfg.gsub! /^: [^\n]*\n/, ''
+        # backup any xml referenced in the configuration.
+        anyconnect_profiles = cfg.scan(Regexp.new('(\sdisk0:/.+\.xml)')).flatten
+        anyconnect_profiles.each do |profile|
+            cfg << (comment profile + "\n" )
+            cmd ("more" + profile) do |xml|
+              cfg << (comment xml)
+            end
+        end
+        # if DAP is enabled, also backup dap.xml
+        if cfg.rindex(/dynamic-access-policy-record\s(?!DfltAccessPolicy)/)
+            cfg << (comment "disk0:/dap.xml\n")
+            cmd "more disk0:/dap.xml" do |xml|
+              cfg << (comment xml)
+              puts xml
```

Comment:
I've tested the model without the 'puts xml' line and it still works so I agree it doesn't seem required.